skada/deep/tests/test_deep_adversarial.py

```diff
@@ -114,6 +115,64 @@ def test_cdan(domain_classifier, domain_criterion, num_feature, max_feature, n_c
     assert history[0]["train_loss"] > history[-1]["train_loss"]
 
 
+@pytest.mark.parametrize(
+    "disc_classifier, disc_criterion, num_features, n_classes",
+    [
+        (
+            DomainClassifier(num_features=10, n_classes=5),
+            CrossEntropyLoss(),
+            10,
+            5,
```

Comment:
Disc_classifier should implement a gradient reversal layer at the beginning to work properly.

Reply:
But it is already implemented in DomainClassifier no? What changes do I have to do in the current PR (knowing that we will open another PR to add a DiscClassifier, decorrelate the domain classifier from the modules and so on)?

Reply:
Yes sorry you have right